src/utils/check.ts

```diff
@@ -1,3 +1,169 @@
+    } catch (error) {
+      console.log(
+        `Serving attempt ${attempt + 1} failed: ${error}. Retrying...`,
+      );
+    }
+  }
+
+  throw `Server failed to start after ${retries} attempts.`;
+};
+
+const build = async (cmd: string) =>
+  new Promise<void>((resolve, reject) => {
+    exec(cmd, (error, stdout, stderr) => {
+      if (error) {
+        return reject(`Build error: ${error}`);
+      }
+      if (stderr) {
+        return reject(`Build stderr: ${stderr}`);
+      }
+      console.log(`Build stdout: ${stdout}`);
+      resolve();
+    });
+  });
+
+const serveBuild = async (cmd: string) =>
+  new Promise<{ serverProcess: any }>((resolve, reject) => {
+    const serverProcess = exec(cmd, (error, stdout, stderr) => {
+      if (error) {
+        return reject(`Serve error: ${error}`);
+      }
+      if (stderr) {
+        return reject(`Serve stderr: ${stderr}`);
+      }
+      console.log(`Serve stdout: ${stdout}`);
+    });
+
+    const url = 'http://localhost:3000/';
+    retry(url, () => resolve({ serverProcess })).catch(reject);
+  });
+
+const stopServer = (serverProcess: any) => {
+  try {
+    process.kill(serverProcess.pid + 1, 'SIGTERM');
+    process.exit(0);
+  } catch (error) {
+    throw `Error stopping the server: ${error}`;
+  }
+};
+
 export async function check(flag: boolean) {
-  console.log(`check${flag ? ' --report' : ''}`);
+  let serverProcess;
+
+  try {
+    const _config = await getConfig();
+    switch (_config?.framework) {
+      case 'React':
+        await build('npm run build');
+        ({ serverProcess } = await serveBuild('npx serve -s build -p 3000'));
+        break;
+      case 'Vue':
+        await build('npx vite build');
+        ({ serverProcess } = await serveBuild('npx serve -s dist -p 3000'));
+        break;
+      case 'Angular':
+        // await build('npx ng build');
+        ({ serverProcess } = await serveBuild('npx ng serve --port 3000'));
+        break;
+      default:
+        throw 'Unsupported framework';
+    }
+
+    const browser = await puppeteer.launch();
+    const page = await browser.newPage();
+    await page.goto(`http://localhost:3000`);
+
+    const results = await new AxePuppeteer(page).analyze();
+
+    let p2 = 0; // Critical and Serious issues
+    let p1 = 0; // Moderate issues
+    let p0 = 0; // Minor issues
+
+    results.violations.forEach((violation) => {
+      violation.nodes.forEach((node) => console.log(node));
+    });
+
+    // Count issues by impact
+    results.violations.forEach((violation) => {
+      switch (violation.impact) {
+        case 'critical':
+        case 'serious':
+          p2 += violation.nodes.length;
+          break;
+        case 'moderate':
+          p1 += violation.nodes.length;
+          break;
+        case 'minor':
+          p0 += violation.nodes.length;
+          break;
+      }
+    });
+
+    // Calculate the raw score
+    const rawScore = ((0.4 * p2 + 0.8 * p1 + p0) / (p1 + p2 + p0)).toFixed(2);
```

Comment:
Are these multipliers e.g. 0.4 from us or AxeScore recommended

Reply:
![image](https://github.com/user-attachments/assets/53dee41d-5e09-4b87-ada5-eaf8a7d52db5)
Yeah it's from axe-core